test/hotspot/jtreg/compiler/loopopts/InvariantCodeMotionReassociateAddSub.java

```diff
@@ -0,0 +1,525 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2.loopopts;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.Utils;
+import java.util.Random;
+
+/*
+ * @test
+ * @bug 8323220
+ * @summary Test loop invariant code motion of add/sub through reassociation
+ * @library /test/lib /
+ * @run main compiler.c2.loopopts.InvariantCodeMotionReassociateAddSub
```

Comment:
You should use `driver` since we do not want to stress the driver VM with additionally passed VM flags like `-Xcomp` when running IR tests.
```suggestion
 * @run driver compiler.c2.loopopts.InvariantCodeMotionReassociateAddSub
```

Reply:
But if we put `driver` here, do the flags get passed to the IR framework test VM?

Reply:
Yes, I think for IR tests, we should not stress the driver VM. The sole purpose of the driver VM is to start the actual test VM (by passing the `javaoptions` and `vmoptions`) and do IR matching afterward. I don't think there is much benefit if we try to also stress the driver VM each time in a normal IR test.

What we could do (and I'm not sure if we already do that) is to have a dedicated (or a set of) IR test that runs with `main` such that the driver VM is also stressed at some point.